src/encoding_implementations/distance_pair_encoding.jl

```diff
@@ -0,0 +1,76 @@
+export PairDistanceEncoding
+
+using Distances: Euclidean
+
+"""
+    PairDistanceEncoding <: Encoding
+    PairDistanceEncoding(min_dist, max_dist; n = 2, metric = Chebyshev(), precise = false)
+
+An encoding that [`encode`](@ref)s points pairs on the form 
+`Tuple{<:AbstractVector, <:AbstractVector}` by first computing their distance 
+using the given `metric`, then dividing the interval [`min_dist, max_dist]` into 
+`n` equal-size bins, and mapping the computed distance onto one of those bins.
+Bins are enumerated as `1:n`. When [`decode`](@ref)-ing the bin integer, the left edge
+of the bin is returned.
+
+If `precise == true`, mapping into bins is more precise at the bin edges. 
+This is slower, so `precise == false` is the default.
+
+## Example
+
+Let's create an example where the minimum and maximum allowed distance is known.
+
+```julia 
+using ComplexityMeasures, Distances, StaticArrays
+m = Chebyshev()
+y = [SVector(1.0), SVector(0.5), SVector(0.25), SVector(0.64)]
+pair1, pair2, pair3 = (y[1], y[2]), (y[2], y[3]), (y[3], y[4])
+dmax = m(pair1...) # dist = 0.50
+dmin = m(pair2...) # dist = 0.25
+dmid = m(pair3...) # dist = 0.39
+
+# This should give five bins with left adges at [0.25], [0.30], [0.35], [0.40] and [0.45]
+encoding = PairDistanceEncoding(dmin, dmax; n = 5, metric = m)
+c1 = encode(encoding, pair1) # 5
+c2 = encode(encoding, pair2) # 1
+c3 = encode(encoding, pair3) # 3
+
+decode(encoding, c3) ≈ [0.35] # true
+```
+"""
+struct PairDistanceEncoding{
+        R, 
+        I <: Integer,
+        MINV <: Real,
+        MAXV <: Real,
+        M} <: Encoding
+    binencoder::R # RectangularBinEncoding
+    n::I
+    min_dist::MINV
+    max_dist::MAXV
+    metric::M
+end
+
+function PairDistanceEncoding(min_dist, max_dist; 
+        n = 2, metric = Chebyshev(), precise = false)
+    bin_edges = range(min_dist, nextfloat(max_dist); length = n + 1)
+    binning = FixedRectangularBinning(bin_edges, 1, precise)
+    binencoder = RectangularBinEncoding(binning)
+    return PairDistanceEncoding(binencoder, n, min_dist, max_dist, metric)
+end
+
+# ----------------------------------------------------------------
+# Pretty printing (see /core/pretty_printing.jl).
+# ----------------------------------------------------------------
+hidefields(::Type{<:PairDistanceEncoding}) = [:binencoder]
```

Comment:
Hm, all this printing code reduces our test coverage. Perhaps in a different PR we can put all the printing ccode in one file and tell the coverage to skip this file from coverage? E.g., this PR got a report of adding lines without coverage.

Reply:
Most of the pretty printing code is in one file. It's very few types that do something custom, so we should just ensure that these lines are tested. In this case, it is just a matter of checking that the string `"binencoder = "` doesn't appear in `repr(SequencialPairDistances())`. I just forgot to add the check.

Reply:
> Perhaps in a different PR we can put all the printing ccode in one file and tell the coverage to skip this file from coverage?

I don't think it is a good idea to exclude printing from testing. Testing the printing code is essential, because any errors can lead to the package being unusable in the REPL without explicitly not printing by terminating each line by `;`.